bookkeeper-server/src/main/java/org/apache/bookkeeper/util/HardLink.java

```diff
@@ -416,6 +471,19 @@ public static void createHardLink(File file, File linkName)
       throw new IOException(
           "invalid arguments to createHardLink: link name is null");
     }
+
+    // if createLink available try first, else fall back to shell command.
+    if (CREATE_LINK_SUPPORTED.get()) {
```

Comment:
We can make the `CREATE_LINK_SUPPORTED` default value true, and remove the test code.
If use createLink failed, change `CREATE_LINK_SUPPORTED` to false.

Reply:
fixed.